src/GDBDebugSession.ts

```
@@ -90,6 +90,17 @@ const ignoreCountRegex = /\s|\>/g;
 const arrayRegex = /.*\[[\d]+\].*/;
 const arrayChildRegex = /[\d]+/;
 
+export function hexToBase64(hex: string): string {
+    if (hex.length % 2 !== 0) {
+        throw new Error('Received memory with incomplete bytes.');
+    }
+    const base64 = Buffer.from(hex, 'hex').toString('base64');
+    if (base64.length === 0 && hex.length !== 0) {
+        throw new Error('Received illformed hex input: ' + hex);
+    }
+    return base64;
+}
```

Looking better! Though the `base64` output string will only be empty if `hex` is empty to begin with.

```ts
export function hexToBase64(hex: string): string {
    if (hex.length !== 0 || hex.length % 2 !== 0) {
        throw new Error('Malformed hex data (empty or incomplete)');
    }
    return Buffer.from(hex, 'hex').toString('base64');
}
```

---

The base64 string will also be empty if the hex is ill-formed (includes characters that aren't hex digits). In that case, the `Buffer.from()` call returns an empty buffer. See the test case that tests `zyxd` and does in fact `throw` due to the current check. It isn't something that should come up, but it would definitely be a problem if it did.

---

See what I mean by things should fail early... This such a weird behavior from `Buffer` to just return empty string if there's an input it doesn't like :/

I might even had misread the check, but I understand now thanks!